contracts/KNS_Retirer_v1.sol

```diff
@@ -146,38 +146,48 @@ contract KNS_Retirer is Initializable, OwnableUpgradeable {
     
     // PUBLIC
 
-    function retireAndKI( uint _USDCAmt ) public {
+    function retireAndKI(
+        uint _USDCAmt,
+        address beneficiary, 
+        string memory domainName
+    ) public {
         require(
             ( _USDCAmt >= 1*(10**6) ), 
             "Payment net of royalties and referrals is less than 1 USDC."
         );
 
         IERC20(USDC).transferFrom(msg.sender, address(this), _USDCAmt);
-        retireBCT((_USDCAmt * retireBCTPercent) / percentFundsReceived );
-        stakeinKI ((_USDCAmt * stakeInKIPercent) / percentFundsReceived );
+
+        retireBCT(
+            (_USDCAmt * retireBCTPercent) / percentFundsReceived,
+            beneficiary,
+            domainName
+        );
         
-        //Check balance after retirement and allocation to auto-retirement KLIMA.
-        uint256 leftoverFunds = IERC20(USDC).balanceOf(address(this));
+        stakeinKI ((_USDCAmt * stakeInKIPercent) / percentFundsReceived );
         
-        IERC20(USDC).approve(treasurer, leftoverFunds);
-        IERC20(USDC).transferFrom(address(this), treasurer, leftoverFunds);
-
+        // transfer the remaining USDC to the treasury address
+        IERC20(USDC).transfer(treasurer, IERC20(USDC).balanceOf(address(this)));
     }
 
     // PRIVATE
 
     //Retires BCT via Klima DAO's Klima Infinity retirement aggregator.
-    function retireBCT(uint _retireAmt) private {
+    function retireBCT(
+        uint _retireAmt, 
+        address beneficiary, 
+        string memory domainName
```

Comment:
Beneficiary address (domain minter) and domain name added as params

Reply:
Noted